File: Java/src/Main/BakedGood.java
package Main;

public abstract class BakedGood {
    private double basePrice;

    public BakedGood(double basePrice) {
        this.basePrice = basePrice;
    }

    public String ToString() {
        return "($" + String.format("%.2f", basePrice) + ")";
    }

    public double getBasePrice() {
        return basePrice;
    }

    public abstract double DiscountedPrice(int quantity);
}

File: Dart/bin/baked_good.dart
abstract class BakedGood {
  final double _basePrice;

  BakedGood(this._basePrice);

  @override
  String toString() {
    return "(\$" + _basePrice.toStringAsFixed(2) + ")";
  }

  double get basePrice => _basePrice;

  double discountedPrice(int quantity);
}
